api/ocm/plugin/ppi/utils.go

```diff
@@ -115,6 +119,111 @@ func (b *nameDescription) Description() string {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+type transferHandler struct {
+	name        string
+	description string
+	questions   []DecisionHandler
+}
+
+func NewTransferHandler(name, desc string) *transferHandler {
+	return &transferHandler{
+		name:        name,
+		description: desc,
+		questions:   nil,
+	}
+}
+
+func (t *transferHandler) GetName() string {
+	return t.name
+}
+
+func (t *transferHandler) GetDescription() string {
+	return t.description
+}
+
+func (t *transferHandler) GetQuestions() []DecisionHandler {
+	return t.questions
+}
+
+func (t *transferHandler) RegisterDecision(h DecisionHandler) error {
+	if TransferHandlerQuestions[h.GetQuestion()] == nil {
+		return errors2.ErrInvalid(KIND_QUESTION, h.GetQuestion())
+	}
+	for _, e := range t.questions {
+		if e.GetQuestion() == h.GetQuestion() {
+			return errors2.ErrAlreadyExists(KIND_QUESTION, e.GetQuestion())
+		}
+	}
+	t.questions = append(t.questions, h)
+	return nil
+}
+
+// DecisionHandlerBase provides access to the
+// non-functional attributes of a DecisionHandler.
+// It can be created with NewDecisionHandlerBase and
+// embedded into the final DecisionHandler implementation.
+type DecisionHandlerBase struct {
+	question    string
+	description string
+	labels      *[]string
+}
+
+func (d *DecisionHandlerBase) GetQuestion() string {
+	return d.question
+}
+
+func (d *DecisionHandlerBase) GetDescription() string {
+	return d.description
+}
+
+func (d *DecisionHandlerBase) GetLabels() *[]string {
+	return d.labels
+}
+
+func NewDecisionHandlerBase(q, desc string, labels ...string) DecisionHandlerBase {
+	return DecisionHandlerBase{q, desc, generics.Pointer(slices.Clone(labels))}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+func ComponentVersionQuestionFunv(f func(p Plugin, question *ComponentVersionQuestion) (bool, error)) func(p Plugin, question Question) (bool, error) {
+	return func(p Plugin, question Question) (bool, error) {
+		return f(p, question.(*ComponentVersionQuestion))
+	}
+}
+
+func ComponentReferenceQuestionFunc(f func(p Plugin, question *ComponentReferenceQuestion) (bool, error)) func(p Plugin, question Question) (bool, error) {
+	return func(p Plugin, question Question) (bool, error) {
+		return f(p, question.(*ComponentReferenceQuestion))
+	}
+}
+
+func ArtifactQuestionFunc(f func(p Plugin, question *ArtifactQuestion) (bool, error)) func(p Plugin, question Question) (bool, error) {
+	return func(p Plugin, question Question) (bool, error) {
+		return f(p, question.(*ArtifactQuestion))
+	}
+}
+
+type defaultDecisionHandler struct {
+	DecisionHandlerBase
+	handler func(p Plugin, question Question) (bool, error)
+}
+
+// NewDecisionHandler provides a default decision handler based on its standard
+// fields and a handler function.
+func NewDecisionHandler(q, desc string, h func(p Plugin, question Question) (bool, error), labels ...string) DecisionHandler {
```

Comment:
If the question types (`ComponentVersionQuestion`, `ComponentReferenceQuestion`, `ArtifactQuestion`, ...) actually implement the `Question` interface and if you define `func(p Plugin, question Question) (bool, error)` as a dedicated type, `ComponentVersionQuestionFunc `, `ComponentReferenceQuestionFunc`, `ArtifactQuestionFunc` would implement that type, which would be quite helpful when trying to implement a `NewDecisionHandler`.

Reply:
introduced type `QuestionResultFunc`